### ISSUE TITLE
alpha removal of HDR png image causes issues with metadata and rendering

### ISSUE THREAD

**Issue by FotisK** [ImageMagick/ImageMagick#7797]
### ImageMagick version

7.1.1-41 Q16-HDRI

### Operating system

MacOS

### Operating system, version and so on

Sequoia 15.1.1

### Description

When trying to remove the (empty) alpha channel from a 16bit, HDR/BT.2020 png file, IM7 does further changes and metadata alterations that affect how the image is eventually rendered in other applications.

### What happens:
After removing the Alpha channel (empty alpha channel) via IM7,  if you open the resulting image in Affinity Photo, macOS preview app or apple photos, the images are washed out and the color profile reads as sRGB instead of BT2020/BT2100PQ
```exiftool``` and ```magick identify -verbose``` indicate that the the icc profile still exists, but there seems to be some warnings and discrepancies. Attached are the relevant outputs of exiftool and magick identify. [diffs.txt](https://github.com/user-attachments/files/17846441/diffs.txt)
file is changed to sRGB instead of of the original BT.2020/BT2100PQ.
Pixel data seem to be relatively unaffected (0,1% difference in the values) though I'm not sure even that should be happening in a lossless process.

### Expected result:
removing the alpha channel should not alter any other metadata and the image should render the same as it did before the removal.

### Sample output
input
![Screenshot 2024-11-21 at 13 39 42](https://github.com/user-attachments/assets/5264b34f-d690-4516-b7bd-97cb2c855936)

output
![Screenshot 2024-11-21 at 13 39 53](https://github.com/user-attachments/assets/744f05be-60d6-4348-b577-2349ecc7b963)

### Steps to Reproduce

1. run ```magick file.png -alpha deactivate fileout.png``` (also tested ```magick file.png -alpha off```, ```magick file.png -alpha remove -alpha off fileout.png```)
2. file should render differently than the original.

### Images

[test.png.zip](https://github.com/user-attachments/files/17846514/test.png.zip)

**Comment by FotisK** [comment]
Thank you @kmilos, very elucidating - I was starting to suspect that something similar was going and now you clarified it for me. There seems to be [a related ongoing discussion in libpng](https://github.com/pnggroup/libpng/pull/626) currently (if I'm not mistaken that is where IM gets its png-capabilities)

**Comment by fmw42** [comment]
I get virtually no difference from 

```
magick test.png -alpha extract alpha.png
magick test.png -alpha off aoff.png
magick compare -metric rmse test.png aoff.png null:
0.343396 (5.23989e-06)
```

On IM 7.1.1-41 on Mac OSX Ventura

![test](https://github.com/user-attachments/assets/8a3a503d-d552-43fa-8294-be5e77ffb638)

![aoff](https://github.com/user-attachments/assets/c2e3e1c7-66e5-4b11-8492-c0fc2194102a)

**Comment by jbowler** [comment]
Magick convert seems to be behaving correctly WRT cICP; it doesn't recognized cICP and the chunk is "unsafe to copy" (last letter of the name) so it must be removed.

The cICP is [9, 16, 0, 1].  The spec is actually here: https://www.itu.int/rec/T-REC-H.273, the latest definition of cICP itself (using the tables in the previous reference) is https://w3c.github.io/png/#cICP-chunk.  Note that the latter is **not** in force yet so subject to change.  I double checked that and it is as @kmilos describes: REC 2020(9) with PQ encoding(16), RGB encoding(0) of a full range image (so with PQ all the values are in the range 0..1).

Unfortunately there is a serious issue here which is part of the problem: magick has added a cHRM chunk which matches REC 709, not REC 2020, and that complete hoses the colours.  This may, in fact, be the primary cause of the washout.  This isn't necessarily magick's fault either; it was told to "remove" the alpha and it looks like it did that by compositing the image on a white background, to do that correctly it has to know the RGB transfer function.

The XMP profile has been retained however it's been converted from a UTF-8 iTXt to a Latin-1 zTXt; that's not going to work if the XMP has multi-byte characters, for example for author names.  The XMP profile repeats the colourspace in [XMP-photoshop] but it says the data is linear.  I don't actually believe the stuff in there; a Canon FDn full frame lens on a micro four thirds Olympus E-M10MarkII?

Supporting cICP though the unknown chunk handling approach is robust and easy; it won't stop working when there is support in libpng so long as png_set_unknown... is called to make the chunks saved as unknown regardless of any support.

**Comment by fmw42** [comment]
May not be an issue, but for IM 7, use magick, not magick convert.

**Comment by svgeesus** [comment]
> The spec is actually here: https://www.itu.int/rec/T-REC-H.273, the latest definition of cICP itself (using the tables in the previous reference) is https://w3c.github.io/png/#cICP-chunk. Note that the latter is **not** in force yet so subject to change.

Recently the 4th edition of H.273 was made publicly available, and the PNG spec updated to reference it:
https://github.com/w3c/png/pull/484

**Comment by FotisK** [comment]
> The XMP profile repeats the colourspace in [XMP-photoshop] but it says the data is linear. I don't actually believe the stuff in there; a Canon FDn full frame lens on a micro four thirds Olympus E-M10MarkII?

@jbowler: fwiw, I can't really comment on the xmp data and the claim that the data is linear, but the lens is indeed the above-mentioned Canon lens from the 80s used with a mount-adapter on the em10m2 body. Of course the value in the field was manually added since the lens has no electronics to pass this piece of information to the body. Also in case that helps, the tiff file is not straight out of the camera but it is the export of an attempt to do some bracketed (HDR) merges using Affinity Photo v2.

**Comment by jbowler** [comment]
@FotisK neat.  I'd naively assumed that most 35mm lenses would have a larger mount which precluded passive adapters for MFT.  Not so; it looks like MFT is (typically?) closer to the film plane so the adapters have sufficient space to "shrink" the mount.  There's an adapter for every interchangeable lens camera I've ever owned.

I use my OM for HDR too.  I use PtGUI, https://ptgui.com/, because so far I've only wanted full 360° images.  It handles the HDR part as well as stitching the images across the dome.  Results are as on PolyHaven, https://polyhaven.com/hdris

I **did not** attempt to check the image to see if it contains HDR data and that invalidates some of what I said.  I'd thought that it might simply be an image with the standard dynamic range (2D, 7EV) converted to PQ.  If your OM has the same software as mine (an OM1) the OM HDR adds 12EV to the dynamic range of the basic 7EV captured by the sensor; so that's 19EV total.  Using OMF (sp?, i.e. TIFF) data this should all work but the 8-bit JPEG output clobbers the dynamic range.  I use the OMF because PtGUI handles it just fine.

I'm finding that producing good test images is something of a nightmare because I'm having to deduce/reverse engineer what each piece of software is doing.  I can write HDR data in PNG without cICP, just using a sufficiently large gamma (small gAMA value) so that's where I'm starting :-)

**Comment by FotisK** [comment]
You are right! I'm still fumbling my way around the peculiarities of each format and software when it comes to extended dynamic range but as you say 8bit jpegs are very limiting especially now that OLED displays have entered our daily lives and the extra dynamic range of the .ORF/RAW could be visually represented and appreciated.
